src/configupdater/option.py

```diff
@@ -107,22 +107,22 @@ def _instantiate_copy(self: T) -> T:
             space_around_delimiters=self._space_around_delimiters,
         )
 
-    def optionxform(self, optionstr: str) -> str:
-        """Delegates :meth:`~configupdater.document.Document.optionxform`
-        to its parent container.
-
-        Please notice that when the option object is :obj:`detached
-        <configupdater.block.Block.detach>`, this method will simply return
-        ``optionstr`` as it is, without any changes.
-        """
-        if self.has_container():
-            section = cast("Section", self.container)
-            return section.optionxform(optionstr)
-        return optionstr
+    @property
+    def section(self) -> "Section":
+        return cast("Section", self.container)
 
     @property
     def key(self) -> str:
-        return self.optionxform(self._key)
+        """Key string associated with the option.
+
+        Please notice that usually, the option key is normalized with
+        :meth:`~configupdater.document.Document.optionxform`, however,
+        when the option object is :obj:`detached <configupdater.block.Block.detach>`,
+        this method will simply return the key as it is, without any changes.
+        """
+        if self.has_container():
+            return self.section.document.optionxform(self._key)
+        return self._key
```

Comment:
Not sure at this point if one should not just raise here in the `else` case, so if no container is there.

Reply:
Hi @FlorianWilhelm thanks for the review.

Quick question here related to your comment: are we ok in raising "NotAttachedError" if someone calls `option.key` on a detached option?

Reply:
I would say so as it simplifies things. The normal use-case would be that a user should not start fiddling around with a detached section or option anyway. Or what do you think?

Reply:
Ok, I will remove and see if the tests fail 😄 

I think that should be the best way of answering this question right? If running the test suite does not require fiddling with `option.key` in a detached state, then it should be fine.

Thanks @FlorianWilhelm

Reply:
Great no errors failing! So changing `__setitem__` in Section was enough to prevent the erros.